packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -90,6 +94,23 @@ export class CdkToolkit {
         throw new Error(`Stack ${stack.name} does not define an environment, and AWS credentials could not be obtained from standard locations or no region was configured.`);
       }
 
+      if (Object.keys(stack.template.Resources || {}).length === 0) { // The generated stack has no resources
+        const cfn = await options.sdk.cloudFormation(stack.environment.account, stack.environment.region, Mode.ForReading);
+        if (!await stackExists(cfn, stack.name)) {
+          warning('%s: stack has no resources, skipping deployment.', colors.bold(stack.name));
+        } else {
+          warning('%s: stack has no resources, skipping deployment and deleting existing stack.', colors.bold(stack.name));
```

Comment:
I'd strip the `skipping deployment and` part here, I fear it might be somewhat confusing. Destroying the stack **is** the deployment here.